File: leetcode/0074.search-a-2D-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for row in matrix:
            i = 0
            j = len(row) - 1
            while(i <= j):
                mid = (i + j) // 2
                if( row[mid] == target ):
                    return True
                elif( row[mid] < target ):
                    i += 1
                else:
                    j -= 1
        return False
        

File: leetcode/0074.search-a-2d-matrix.ts
function searchMatrix(matrix: number[][], target: number): boolean {
    const r = matrix.length;
    const c = matrix[0].length;

    let low = 0;
    let high = (r * c) - 1;

    while (low <= high) {
        const mid = Math.floor((low + high) / 2);
        const midR = Math.floor(mid / c);
        const midC = mid % c;

        if (matrix[midR][midC] === target) return true;
        else if (matrix[midR][midC] < target) low = mid + 1;
        else high = mid - 1;
    }

    return false;
};